urllib3/response.py

```diff
@@ -56,14 +56,30 @@ class GzipDecoder(object):
 
     def __init__(self):
         self._obj = zlib.decompressobj(16 + zlib.MAX_WBITS)
+        self._state = 'first'
 
     def __getattr__(self, name):
         return getattr(self._obj, name)
 
     def decompress(self, data):
+        ret = binary_type()
+        if self._state == 'error':
+            return ret
         if not data:
             return data
-        return self._obj.decompress(data)
+        try:
+            ret = self._obj.decompress(data)
+            while self._obj.unused_data:
+                self._state = 'other'
+                (self._obj, data) = (zlib.decompressobj(16 + zlib.MAX_WBITS),
```

Comment:
There is no reason to shoehorn these items onto one line. Please separate them onto their own lines for better readability.

Reply:
Done.